c10/core/TensorImpl.h

```diff
@@ -2238,14 +2238,16 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
         const auto dim_ = dim();
         sizes_and_strides_.resize(dim_);
         if (dim_ > 0) {
+          bool overflowed = false;
           const auto last_idx = dim_ - 1;
           sizes_and_strides_.stride_at_unchecked(last_idx) = 1;
           for (auto i = last_idx - 1; i >= 0; --i) {
-            sizes_and_strides_.stride_at_unchecked(i) =
-                sizes_and_strides_.stride_at_unchecked(i + 1) *
-                std::max<int64_t>(
-                    sizes_and_strides_.size_at_unchecked(i + 1), 1);
+            overflowed |= c10::mul_overflows(sizes_and_strides_.stride_at_unchecked(i + 1), std::max<int64_t>(
+                    sizes_and_strides_.size_at_unchecked(i + 1), 1), std::addressof(sizes_and_strides_.stride_at_unchecked(i)));
+            TORCH_CHECK(!overflowed,
+              "Stride calculation overflowed with stride of dim", i);
```

Comment:
Is it better to test this only once outside of the loop?

Reply:
Yes. Moved it outside the loop.